File: csharp/Original/PublishingHouse.Infrastructure/Aggregates/Aggregate.cs
namespace PublishingHouse.Core.Aggregates;

public abstract class Aggregate<TKey>
{
    public TKey Id { get; protected set; }

    private List<IDomainEvent> _domainEvents = new();

    protected Aggregate(TKey id)
    {
        Id = id;
    }

    protected void AddDomainEvent(IDomainEvent domainEvent)
    {
        _domainEvents.Add(domainEvent);
    }

    public void ClearEvents()
    {
        _domainEvents.Clear();
    }

    public IReadOnlyCollection<IDomainEvent> DomainEvents => _domainEvents.AsReadOnly();
}

File: java/src/main/java/io/eventdriven/slimdownaggregates/shorter/original/core/Aggregate.java
package io.eventdriven.slimdownaggregates.shorter.original.core;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.UUID;

public abstract class Aggregate {
  protected UUID id;
  private List<IDomainEvent> domainEvents;

  protected Aggregate(UUID id) {
    this.id = id;
    this.domainEvents = new ArrayList<>();
  }

  protected void addDomainEvent(IDomainEvent domainEvent) {
    this.domainEvents.add(domainEvent);
  }

  public void clearEvents() {
    this.domainEvents.clear();
  }

  public List<IDomainEvent> getDomainEvents() {
    return Collections.unmodifiableList(domainEvents);
  }

  public UUID getId() {
    return id;
  }
}

